File: app/src/main/java/com/refugio/pawrescue/ui/publico/PublicAnimalsAdapter.java
package com.refugio.pawrescue.ui.publico;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;
import androidx.recyclerview.widget.ListAdapter;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;
import com.refugio.pawrescue.R;
import com.refugio.pawrescue.model.Animal;
import com.refugio.pawrescue.databinding.ItemAnimalPublicBinding;

public class PublicAnimalsAdapter extends ListAdapter<Animal, PublicAnimalsAdapter.AnimalViewHolder> {

    private final OnAnimalClickListener listener;

    public interface OnAnimalClickListener {
        void onAnimalClick(Animal animal);
    }

    public PublicAnimalsAdapter(OnAnimalClickListener listener) {
        super(DIFF_CALLBACK);
        this.listener = listener;
    }

    private static final DiffUtil.ItemCallback<Animal> DIFF_CALLBACK = new DiffUtil.ItemCallback<Animal>() {
        @Override
        public boolean areItemsTheSame(@NonNull Animal oldItem, @NonNull Animal newItem) {
            return oldItem.getIdAnimal() != null && oldItem.getIdAnimal().equals(newItem.getIdAnimal());
        }

        @Override
        public boolean areContentsTheSame(@NonNull Animal oldItem, @NonNull Animal newItem) {
            return oldItem.getNombre().equals(newItem.getNombre()) &&
                    oldItem.getRaza().equals(newItem.getRaza()) &&
                    oldItem.getEstadoRefugio().equals(newItem.getEstadoRefugio());
        }
    };

    @NonNull
    @Override
    public AnimalViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemAnimalPublicBinding binding = ItemAnimalPublicBinding.inflate(
                LayoutInflater.from(parent.getContext()),
                parent,
                false
        );
        return new AnimalViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull AnimalViewHolder holder, int position) {
        Animal animal = getItem(position);
        holder.bind(animal);
    }

    class AnimalViewHolder extends RecyclerView.ViewHolder {
        private final ItemAnimalPublicBinding binding;

        public AnimalViewHolder(ItemAnimalPublicBinding binding) {
            super(binding.getRoot());
            this.binding = binding;

            binding.getRoot().setOnClickListener(v -> {
                int position = getAdapterPosition();
                if (position != RecyclerView.NO_POSITION && listener != null) {
                    listener.onAnimalClick(getItem(position));
                }
            });
        }

        public void bind(Animal animal) {
            // Nombre
            binding.tvNombre.setText(animal.getNombre());

            // Raza
            binding.tvRaza.setText(animal.getRaza());

            // Edad aproximada (tu modelo usa edadAprox que es String: "Cachorro", "Adulto")
            String edadTexto = animal.getEdadAprox() != null ? animal.getEdadAprox() : "Edad desconocida";
            binding.tvEdad.setText(edadTexto);

            // Sexo
            binding.tvSexo.setText(animal.getSexo());

            // Estado (tu modelo usa estadoRefugio: "Rescatado", "Disponible Adopcion", "Adoptado")
            if ("Disponible Adopcion".equals(animal.getEstadoRefugio()) ||
                    "Disponible".equals(animal.getEstadoRefugio())) {
                binding.tvEstado.setText("❤️");
                binding.tvEstado.setBackgroundResource(R.drawable.bg_badge_available);
            } else if ("Adoptado".equals(animal.getEstadoRefugio())) {
                binding.tvEstado.setText("✓");
                binding.tvEstado.setBackgroundResource(R.drawable.bg_badge_adopted);
            } else {
                binding.tvEstado.setText("🐾");
                binding.tvEstado.setBackgroundResource(R.drawable.bg_badge_adopted);
            }

            // Cargar imagen con Glide
            if (animal.getFotoUrl() != null && !animal.getFotoUrl().isEmpty()) {
                Glide.with(binding.getRoot().getContext())
                        .load(animal.getFotoUrl())
                        .placeholder(R.drawable.bg_gradient_green)
                        .error(R.drawable.bg_gradient_green)
                        .centerCrop()
                        .into(binding.ivAnimalPhoto);
            } else {
                binding.ivAnimalPhoto.setImageResource(R.drawable.bg_gradient_green);
            }
        }
    }
}

File: app/src/main/java/com/refugio/pawrescue/ui/theme/public_user/PublicAnimalsAdapter.kt
package com.refugio.pawrescue.ui.theme.public_user

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.refugio.pawrescue.R
import com.refugio.pawrescue.data.model.Animal
import com.refugio.pawrescue.databinding.ItemPublicAnimalBinding

class PublicAnimalsAdapter(
    private val onAnimalClick: (Animal) -> Unit
) : ListAdapter<Animal, PublicAnimalsAdapter.AnimalViewHolder>(AnimalDiffCallback()) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AnimalViewHolder {
        val binding = ItemPublicAnimalBinding.inflate(
            LayoutInflater.from(parent.context),
            parent,
            false
        )
        return AnimalViewHolder(binding)
    }

    override fun onBindViewHolder(holder: AnimalViewHolder, position: Int) {
        holder.bind(getItem(position))
    }

    inner class AnimalViewHolder(
        private val binding: ItemPublicAnimalBinding
    ) : RecyclerView.ViewHolder(binding.root) {

        fun bind(animal: Animal) {
            binding.apply {
                tvNombre.text = animal.nombre
                tvInfo.text = "${animal.raza} • ${getEdadText(animal.edad)}"

                // Características
                val caracteristicas = mutableListOf<String>()
                if (animal.vacunasCompletas) caracteristicas.add("✓ Vacunado")
                if (animal.esterilizado) caracteristicas.add("✓ Esterilizado")
                if (animal.desparasitado) caracteristicas.add("✓ Desparasitado")
                tvCaracteristicas.text = caracteristicas.joinToString(" • ")

                // Emoji según especie
                tvEmoji.text = when (animal.especie) {
                    "perro" -> "🐕"
                    "gato" -> "🐈"
                    "ave" -> "🐦"
                    else -> "🐾"
                }

                // Cargar foto si existe
                if (animal.fotosPrincipales.isNotEmpty()) {
                    Glide.with(itemView.context)
                        .load(animal.fotosPrincipales.first())
                        .placeholder(R.drawable.bg_gradient_green)
                        .into(ivAnimalPhoto)
                    tvEmoji.visibility = android.view.View.GONE
                } else {
                    ivAnimalPhoto.setImageResource(R.drawable.bg_gradient_green)
                    tvEmoji.visibility = android.view.View.VISIBLE
                }

                root.setOnClickListener { onAnimalClick(animal) }
            }
        }

        private fun getEdadText(edad: String): String {
            return when (edad) {
                "cachorro" -> "Cachorro"
                "adulto" -> "Adulto"
                "vejez" -> "Mayor"
                else -> edad
            }
        }
    }

    private class AnimalDiffCallback : DiffUtil.ItemCallback<Animal>() {
        override fun areItemsTheSame(oldItem: Animal, newItem: Animal): Boolean {
            return oldItem.id == newItem.id
        }

        override fun areContentsTheSame(oldItem: Animal, newItem: Animal): Boolean {
            return oldItem == newItem
        }
    }
}